### PR TITLE
Keep `qtpy` in headless tests and fix resulting failure

### PR DESCRIPTION
**Pull request by DragaDoncila**
# Description

This pull request updates the `tox` configuration for our headless tests to keep `qtpy` in the environment when running tests, and fixes the resulting failure by updating the `_safe_register_qt_actions` function to catch a more generic `ImportError`. 

As we can see in [this test run](https://github.com/napari/napari/actions/runs/8382772241/job/22957514533?pr=6764), if we keep `qtpy` in the environment when running headless tests (mimicking what a user would have if they ran `pip install napari`), tests initially fail because our code was catching a more specific `ModuleNotFoundError`. Users should not have to explicitly uninstall a required dependency to make use of `headless` mode. Now with a more generic `ImportError`, headless is available with or without `qtpy` in the environment.

# References

See [the npe2 PR](https://github.com/napari/npe2/pull/344) where this was first observed for more detail.

### CODE REVIEW

**Comment by Czaki** [comment]
The lack of reproducibility is explained here.

**Comment by DragaDoncila** [comment]
Should we add another separate test for headless without qtpy? I don't really see the relevance of it tbh but if that's something we "guarantee" via docs or via prior expectation then maybe.

**Comment by Czaki** [comment]
We test without qtpy. The problem is npe2 runs test with qtpy

**Comment by DragaDoncila** [comment]
>  We test without qtpy. The problem is npe2 runs test with qtpy

As I mentioned in the description of this PR and the comment in npe2, we **should** be testing with qtpy. Users should not have to uninstall qtpy explicitly to make use of headless mode.

File: napari/plugins/_npe2.py
Comment (jni) [inline napari/plugins/_npe2.py:368]:
```suggestion
    except ImportError:  # pragma: no cover
        # if no Qt bindings are installed (PyQt/PySide), then trying to import
        # qtpy will raise an ImportError, *not* a ModuleNotFoundError
```

**Review by jni — APPROVED** [review]
Great, thanks for the investigation @aganders3 and for the fix @DragaDoncila! (Which I totally agree with now that we know the causes.)

**Comment by jni** [comment]
@Czaki I think this is the correct fix — we should not test without all our requirements — it does not make any more sense than testing after uninstalling NumPy.

**Comment by jni** [comment]
Because the failing CI on npe2 is holding up a bugfix release, I'm going to merge this so as to unblock the npe2 release. Thanks to everyone who helped here! 🙏

**Pull request merged**
